torrent.go

```diff
@@ -1,13 +1,110 @@
 package tpb
 
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+)
+
 // Torrent represents a Torrent
 type Torrent struct {
-	Name        string
-	Leechers    int
-	Seeders     int
-	User        string
-	Magnet      string
-	Size        uint64
-	Category    TorrentCategory
-	SubCategory TorrentCategory
+	ID          int             `json:"id"`
+	Category    TorrentCategory `json:"category"`
+	Status      UserStatus      `json:"status"`
+	Name        string          `json:"name"`
+	NumFiles    int             `json:"num_files"`
+	Size        uint64          `json:"size"`
+	Seeders     int             `json:"seeders"`
+	Leechers    int             `json:"leechers"`
+	User        string          `json:"username"`
+	Added       time.Time       `json:"added"`
+	Description string          `json:"descripiton"`
+	InfoHash    string          `json:"info_hash"`
+	ImdbID      string          `json:"imdb"`
+	// Magnet      string          `json:"magnet"`
+}
+
+// GetMagnet returns a Magnet for a torrent
+func (t *Torrent) GetMagnet() string {
+	var trackers = "tr=udp%3A%2F%2Ftracker.coppersurfer.tk%3A6969%2Fannounce&tr=udp%3A%2F%2F9.rarbg.to%3A2920%2Fannounce&tr=udp%3A%2F%2Ftracker.opentrackr.org%3A1337&tr=udp%3A%2F%2Ftracker.internetwarriors.net%3A1337%2Fannounce&tr=udp%3A%2F%2Ftracker.leechers-paradise.org%3A6969%2Fannounce&tr=udp%3A%2F%2Ftracker.coppersurfer.tk%3A6969%2Fannounce&tr=udp%3A%2F%2Ftracker.pirateparty.gr%3A6969%2Fannounce&tr=udp%3A%2F%2Ftracker.cyberia.is%3A6969%2Fannounce"
```

Comment:
Maybe the tracker list should be a global var somewhere

Reply:
I changed the way we generate the magnet